File: array/two-sum.js
/**
 * @param {number[]} nums
 * @param {number} target
 * @return {number[]}
 */
var twoSum = function(nums, target) {
    charInd = {}
    for(let i = 0; i < nums.length; i++) {
        charInd[nums[i]] = i
    }
    console.log(charInd)
    for(let i = 0; i < nums.length; i++) {
        if(i != charInd[target-nums[i]] && target - nums[i] in charInd) {
            console.log(target-nums[i])
            return [i, charInd[target - nums[i]]]
        }
    }
};

File: array/two-sum.py
class Solution:
    def twoSum(self, nums: List[int], target: int) -> List[int]:
        # key: num val: ind
        # match = target - num and ind != curr 
        # hashmap for o(1) lookup o(n) space
        n_ind = {}
        for i in range(len(nums)):
            match = target - nums[i]
            curr = nums[i]
            if match in n_ind:
                return [i, n_ind[match]]
            n_ind[curr] = i
        return [-1,-1]